### DISCUSSION TITLE
Single-threading with capsules not working

### DISCUSSION THREAD

**Discussion by awei82** [sidekiq/sidekiq#5911] [Q&A] [answered]
Hi, I'm upgrading my Sidekiq version from 6.5 to 7.1, and I'm running into an issue implementing [single-threading  Capsules](https://github.com/sidekiq/sidekiq/wiki/Advanced-Options#capsules). I was previously using the [sidekiq-limit_fetch](https://github.com/deanpcmad/sidekiq-limit_fetch) library to limit concurrency for a queue, but it looks like Capsules now support this functionality.

This is a POC showing what I'm running into (ruby version 3.2.2, sidekiq version 7.1)
I defined a 'unsafe' queue that has a concurrency of 1, and assign a job to that queue:
``` yml
# sidekiq.yml
---
:queues:
  - default
  - unsafe
```
``` ruby
# job.rb
require 'sidekiq'

Sidekiq.configure_server do |config|
  # https://github.com/sidekiq/sidekiq/wiki/Advanced-Options#capsules
  config.capsule('unsafe') do |cap|
    cap.concurrency = 1
    cap.queues = %w[unsafe]
  end
end

class TestJob
  include Sidekiq::Job
  sidekiq_options queue: 'unsafe'

  # Print the numbers 1..count with one second delay between each number
  def perform(count)
    (1..count).each do |i|
      puts i
      sleep 1
    end
  end
end
```
command to run the sidekiq job:
`bundle exec sidekiq -r ./job.rb -C ./sidekiq.yml`
testing the job:
``` ruby
# irb
require './job'

5.times do
  TestJob.perform_async(10)
end
```

I want the jobs to run sequentially, not concurrently. Am I doing something wrong here?

Thanks for the help!

**Comment by mperham** [comment, marked as answer]
Well, your config lists "unsafe" as a queue to process by default so the default capsule is picking up unsafe jobs too. Remove it from your config yml.

**Comment by awei82** [comment]
Oh, I see. I mistakenly thought that the queues had to be defined in the config file before being referenced in the capsule configuration.

I deleted the config file, and it's working now. Thanks @mperham !
